### PR TITLE
v0 burn compat 

### PR DESCRIPTION
**Pull request by tok-kkk**
- Compat layer for v0 burns
- Use connection pool for redis interaction 
- Fetch public key and max confirmations from darknode when booting
- Fix incorrect `amount` field for v0 compat mint

### CODE REVIEW

File: resolver/resolver.go [hunk 1]
```diff
@@ -72,11 +72,6 @@ func (resolver *Resolver) QueryBlocks(ctx context.Context, id interface{}, param
...
 
 func (resolver *Resolver) SubmitTx(ctx context.Context, id interface{}, params *jsonrpc.ParamsSubmitTx, req *http.Request) jsonrpc.Response {
-	// When a v0 burn tx gets submitted via RPC, we have to wait for the watcher to detect it before submitting
-	// because it does not have sufficient data to create a valid v1 tx hash
-	// (it just contains a ref to the burn event height + the v0 selector,
-	// and the contract doesn't have a way to query by event height, and can't really filter either)
-	//
 	// As such, we will just respond with the v0 hash so that renjs-v1 can continue as normal, but
 	// we won't actually submit to the darknodes
 	emptyParams := jsonrpc.ParamsSubmitTx{}
```
Comment (jazg) [inline resolver/resolver.go:77]: Should we get rid of this logic now since we are able to submit to Darknodes?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 0.4.5
+- Compat layer for v0 burns
+- Use connection pool for redis interaction 
+- Fetch public key and max confirmations from darknode when booting
+- Fix incorrect `amount` field for v0 compat mint
```
Comment (jazg) [inline CHANGELOG.md:5]:
```suggestion
- Update compatibility layer for v0 burns
- Use connection pool for Redis interaction
- Fetch public key and max confirmations from Darknode when booting
- Fix incorrect `amount` field for v0 mints
```

**Review by jazg — CHANGES_REQUESTED** [review]
Nice stuff, looks good overall! Just requested a few changes.

File: compat/v0/compat.go [hunk 3]
```diff
@@ -357,6 +361,12 @@ func TxFromV1Tx(t tx.Tx, hasOut bool, bindings binding.Bindings) (Tx, error) {
 		}
 	}
 
+	tx.Autogen.Set(Arg{
+		Name:  "sighash",
+		Type:  "b32",
+		Value: B32(sighash),
+	})
```
[the remark this answers is not in the document]
Comment (tok-kkk) [reply compat/v0/compat.go:368]: Yes, I checked with Noah, empty sighash should be fine for old ren-js version.

File: lightnode.go [hunk 4]
```diff
@@ -141,6 +138,35 @@ func New(options Options, ctx context.Context, logger logrus.FieldLogger, sqlDB
 	)
 
 	watchers := map[multichain.Chain]map[multichain.Asset]watcher.Watcher{}
+
+	// solClient := solanaRPC.NewClient(bindingsOpts.Chains[multichain.Solana].RPC.String())
+	// for _, selector := range options.Whitelist {
```
Comment (TheAncientGoat) [inline lightnode.go:143]: Is there a reason we have disabled watchers?

File: resolver/resolver.go [hunk 5]
```diff
@@ -74,23 +74,6 @@ func (resolver *Resolver) QueryBlocks(ctx context.Context, id interface{}, param
 }
 
 func (resolver *Resolver) SubmitTx(ctx context.Context, id interface{}, params *jsonrpc.ParamsSubmitTx, req *http.Request) jsonrpc.Response {
-	// When a v0 burn tx gets submitted via RPC, we have to wait for the watcher to detect it before submitting
```
Comment (TheAncientGoat) [inline resolver/resolver.go:77]: Have we tested burn submission via ren-js?

File: lightnode.go (hunk 4 above)
Comment (tok-kkk) [reply lightnode.go]: I have removed this commented code, it was from resolving some conflicts. The watcher initialization happens below this commented code.

File: resolver/resolver.go (hunk 5 above)
Comment (tok-kkk) [reply resolver/resolver.go:77]: No. Would be good if you could help test with ren-js

**Review by jazg — APPROVED** [review]
Happy for this to be merged assuming there is sufficient testing using the local/staging environments.

**Pull request merged**